src/dbus_proxy/_proxy.lua

```diff
@@ -175,6 +192,45 @@ local function call(proxy, interface, method, args)
   return result
 end
 
+--- Asynchronously call a method with arguments from a given interface on a proxy object.
+-- @param[type=Proxy] proxy a Proxy object
+-- @param[type=string] interface the interface name
+-- @param[type=string] method the method name
+-- @param[type=function] user_callback a callback to be called upon response receival
+-- @param[type=any] user_data context data, will be passed back on callback call
+-- @param[type=table] args an array of tables that have the `type` and `value` fields.
+-- For example
+--
+--    {
+--      {type = "s", value = "a string"},
+--      {type = "v", value = lgi.GLib.Variant("s", "a string variant")}
+--    }
+--
+local function call_async(proxy, interface, method, user_callback, user_data, args)
```

Comment:
I'm not convinced about naming the parameter `user_data`, maybe I'm not familiar with the term, but I think that `context` might be more appropriate.